### ISSUE TITLE
CPR doesn't work

### ISSUE THREAD

**Issue by SlimF** [relrelb/CPPS-PCL#29]
I log in with my credentials, I see that the connection has been established but I cannot input any commands. Tried both on W10 and Arch Linux

```
> Connecting to login server at server.cprewritten.net:6112...
> Logging in...
> Sending verChk request...
> # Send: <msg t="sys"><body action="verChk" r="0"><ver v="153" /></body></msg>
> # Receive: <msg t="sys"><body action="apiOK" r="0"></body></msg>
> Received apiOK response
> Sending rndK request...
> # Send: <msg t="sys"><body action="rndK" r="-1"></body></msg>
> # Receive: <msg t="sys"><body action="rndK" r="-1"><k>a94c5ed2140fc249ee3ce0729e19af5a</k></body></msg>
> Received rndK response: a94c5ed2140fc249ee3ce0729e19af5a
> # Send: <msg t="sys"><body action="login" r="0"><login z="w1"><nick><![CDATA[tashak97]]></nick><pword><![CDATA[65f53eb9a9487741b171341fbdce7dd8]]></pword></login></body></msg>
> # Receive: %xt%gs%-1%100,4|101,0|106,0|200,0|201,0|202,0|203,0|204,0|205,0|206,0|207,0|208,0|209,0|210,0|211,0|1337,0|8888,0|8889,0|9993,0%
> # Receive: %xt%ssa%5022150%AodPa6_aQPEt4j24bQ6LkMoLqbj2uAnQ%{2ba370fa-1d2e-93c7-5e35-4bae1792d157}%
> # Receive: %xt%l%-1%5022150%WZkYRbzFQYWs7d.SrMhgh8dg7C2rB-%%100,4|101,0|106,0|200,0|201,0|202,0|203,0|204,0|205,0|206,0|207,0|208,0|209,0|210,0|211,0|1337,0|8888,0|8889,0|9993,0%
> Logged in
> Connecting to game server at server.cprewritten.net:7001...
> Joining server...
> Sending verChk request...
> # Send: <msg t="sys"><body action="verChk" r="0"><ver v="153" /></body></msg>
> # Receive: <msg t="sys"><body action="apiOK" r="0"></body></msg>
> Received apiOK response
> Sending rndK request...
> # Send: <msg t="sys"><body action="rndK" r="-1"></body></msg>
> # Receive: <msg t="sys"><body action="rndK" r="-1"><k>3Y3rjFV6A</k></body></msg>
> Received rndK response: 3Y3rjFV6A
> # Send: <msg t="sys"><body action="login" r="0"><login z="w1"><nick><![CDATA[tashak97]]></nick><pword><![CDATA[d7c4531e78b6e40eef2638fef0c251bfWZkYRbzFQYWs7d.SrMhgh8dg7C2rB-]]></pword></login></body></msg>
> # Receive: %xt%l%-1%Club Penguin Rewritten%
> # Send: %xt%s%j#js%-1%5022150%WZkYRbzFQYWs7d.SrMhgh8dg7C2rB-%en%
```

**Comment by missaustraliana** [comment]
Same Issue,
But I Know What Is Causing This.
It's Because CPR's New Security
So Here Is What Is Stopping The Connection From Being Sent.
So CPR has better bot and client protection and therefore kicks you from the game because the tool  is a "client"
Hope There Is A Solution Soon.

Thanks,
Acid

**Comment by SlimF** [comment]
Thank you for the explanation!! Let us hope

**Comment by missaustraliana** [comment]
Anytime!

Yes, We Will Praying That @relrelb Bypasses the Security Soon.

Thanks,
Acid

**Comment by relrelb** [comment]
Thank you all for catching this issue.

It seems to be caused by a minor change that CPR made to the game protocol (isn't kind of security mitigation).
I already have a work-in-progress patch that solves it, hopefully I'll sort it out when I have time.

**Comment by relrelb** [comment]
Fixed support in commit 64d7a6644b6a0d44aa9daff35d8f2b5ddf02dded and it worked locally for me, so closing the issue.

**Comment by SlimF** [comment]
Yep, working for me too! Thank you!!
